app/pods/search-schema/create/route.js

```diff
@@ -0,0 +1,49 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+    model(params) {
+        return this.explorer.getCluster(params.clusterId, this.store);
+    },
+
+    actions: {
+        createSchema: function(clusterId, schemaName, schemaContent) {
+            let self = this;
+            let xmlDoc = null;
+            let url = `/riak/clusters/${clusterId}/search/schema/${schemaName}`;
+
+            try {
+                xmlDoc = Ember.$.parseXML(schemaContent);
+            } catch(error) {
+                // TODO: Put in proper error messaging
+                alert('Invalid XML. Please check and make sure schema is valid xml.');
+                return;
+            }
+
+            if (!Ember.$(xmlDoc).find('schema').attr('name')) {
+                // TODO: Put in proper error messaging
+                alert('Solr requires that the schema tag has a name attribute. Please update your xml.');
+                return;
+            }
+
+            if (!Ember.$(xmlDoc).find('schema').attr('version')) {
+                // TODO: Put in proper error messaging
+                alert('Solr requires that the schema tag has a version attribute. Please update your xml.');
+                return;
+            }
+
+            return Ember.$.ajax({
+                type: 'PUT',
+                url: url,
+                contentType: 'application/xml',
+                processData: false,
+                data: xmlDoc
+            }).then(function(data) {
+                alert('schema created!');
```

Comment:
> I don't know how to associate this schema to a an index

No worries, that bit is easier -- it will be handled in #23. The only way to associate a schema with an index is to send an 'edit index' request to riak, and give it the schema id. 

> I could create a new schema in ED right here and then transition to that route, but once you leave there would be no way to get back to that view without manually typing in the url.

Absolutely agree, currently this is awkward. But, all of that will be solved with https://github.com/basho-labs/riak_explorer/issues/89. Nothin' really to do with it now, except to wait.

By the way, lest you think this is just an Explorer limitation -- this is basically a fundamental problem with Riak Search in general, currently. You can upload a schema... but there's no way to list which schemas you've already uploaded. So you basically have to remember what you did. And then use that knowledge to edit indexes. Madness, eh?